templates/CRM/Contribute/Form/Contribution/Main.tpl

```diff
@@ -454,6 +454,17 @@
 
     $('#pricevalue').each(toggleBillingBlockIfFree).on('change', toggleBillingBlockIfFree);
   });
+
+  CRM.$(function($) {
+    // Update pledge contribution amount when pledge checkboxes change
+    cj("input[name^='pledge_amount']").on('change', function() {
+      var total = 0;
+      cj("input[name^='pledge_amount']:checked").each(function() {
+        total += Number(cj(this).attr('amount'));
+      });
+      cj("input[name^='price_']").val(total.toFixed(2));
```

Comment:
That would be a bigger refactoring job. In order to change `cj` to `$` you need to have the `$` variable in scope. Your wrapper `CRM.$(function($) {` provides that. But looks like the other js in this tpl is unscoped. Def needs fixing, but is a bit tedious - have to decide whether each function really *needs* to be global & how it should be wrapped. Feel free to chip away at it if that all makes sense to you.